### DISCUSSION TITLE
Problem synchronous Widget

### DISCUSSION THREAD

**Discussion by AlaHajBrahim** [eclipse-theia/theia#14024] [Q&A] [answered]
![Screenshot 2024-08-08 114718](https://github.com/user-attachments/assets/d7707a55-5511-4c4a-a6e7-90117e28e3d3)

**Comment by msujew** [comment, marked as answer]
Hey @AlaHajBrahim,

the `@postConstruct()` decorated method is not allowed to return a promise. As far as I can tell, your `init()` method isn't actually async - you can just make it return `void` and remove the `async` modifier to fix this.

**Reply by AlaHajBrahim** [reply]
Thank you
